src/main/java/org/example/bookweb/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+import org.example.bookweb.repository.order.OrderItemRepository;
+import org.example.bookweb.repository.order.OrderRepository;
+import org.example.bookweb.repository.shopping.cart.ShoppingCartRepository;
+import org.example.bookweb.service.shopping.cart.ShoppingCartService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartService shoppingCartService;
+    private final UserRepository userRepository;
+
+    @Override
+    public Order createNewOrder(User user, OrderRequestDto orderRequestDto) {
+        Order newOrder = new Order();
+        newOrder.setUser(user);
+        newOrder.setShippingAddress(orderRequestDto.getShippingAddress());
+        newOrder.setStatus(Order.Status.NEW);
+        newOrder.setOrderDate(LocalDateTime.now());
+        newOrder.setTotal(BigDecimal.ZERO);
+        return orderRepository.save(newOrder);
+    }
+
+    @Transactional
+    @Override
+    public OrderResponseDto placeOrder(User user, OrderRequestDto orderRequestDto) {
+        ShoppingCart userCart = shoppingCartRepository.findShoppingCartByUserId(user.getId());
+        User userDb = userRepository.findById(user.getId()).get();
+        Order newSavedOrder = createNewOrder(userDb, orderRequestDto);;
+
+        Set<OrderItem> orderItems = userCart.getCartItems().stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = orderItemMapper.toOrderItem(cartItem);
+                    orderItem.setPrice(cartItem.getBook().getPrice()
+                            .multiply(BigDecimal.valueOf(cartItem.getQuantity())));
+                    orderItem.setOrder(newSavedOrder);
+                    return orderItem;
+                }).collect(Collectors.toSet());
+        orderItemRepository.saveAll(orderItems);
+
+        newSavedOrder.setTotal(
+                orderItems.stream()
+                .map(OrderItem::getPrice)
+                .reduce(BigDecimal.ZERO, BigDecimal::add)
+        );
+        newSavedOrder.setOrderItems(orderItems);
+
+        shoppingCartRepository.deleteById(userCart.getId());
+        shoppingCartService.createShoppingCart(userDb);
+        return orderMapper.toDto(newSavedOrder);
+    }
+
+    @Override
+    public List<OrderResponseDto> getOrdersByUser(User user) {
+        return orderMapper.toDto(orderRepository.findByUserId(user.getId()));
+    }
+
+    @Override
+    public OrderResponseDto updateStatusByOrderId(Long orderId,
+                                                  OrderStatusUpdateRequest statusUpdateRequest) {
+        Order.Status statusEnum = Order.Status.fromString(statusUpdateRequest.getStatus());
```

Comment:
```suggestion
        Order order = orderRepository.findById(orderId)
                .map(o -> {
                    o.setStatus(Status.valueOf(orderDto.status()));
                    return o;
                })
                .orElseThrow(() -> new EntityNotFoundException(
                        String.format("Order with id: %d not found", orderId)
                ));
        return orderMapper.toUpdateDto(orderRepository.save(order));
```

Where 
```
@Mapping(target = "orderId", source = "id")
UpdateOrderResponseDto toUpdateDto(Order order);
```

So you don't need fromString() method